features/login/impl/src/main/kotlin/io/element/android/features/login/impl/accountprovider/AccountProviderDataSource.kt

```diff
@@ -18,21 +19,33 @@ import javax.inject.Inject
 
 @SingleIn(AppScope::class)
 class AccountProviderDataSource @Inject constructor(
-    enterpriseService: EnterpriseService,
+    enterpriseService: EnterpriseService? = null,
```

Comment:
Better to avoid using `null` in injected constructor parameters. This is not a common practice, even if it seems to compile/work fine.